### PR TITLE
CloudWatch: Display dynamic label field in case feature is enabled

### LINKED ISSUE #48613: CloudWatch: Display dynamic label field in case feature is enabled
Part of https://github.com/grafana/grafana/issues/48434

In case dynamic labels feature toggle is enabled, the label field should be displayed in the metrics query editor. If not, the alias field should be displayed.

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:

This PR makes sure the new label field is rendered in case the dynamic label feature is enabled. If not enabled, the legacy alias field is rendered. Also moves all components related to the MetricsQueryEditor to its own folder to make it easier to navigate in this part of the frontend code. 

**Which issue(s) this PR fixes**:

Fixes #48613
Part of https://github.com/grafana/grafana/issues/48434

**Special notes for your reviewer**:
Currently the label field is just an input with a fixed width, so in case a long expression is used all text might not be visible. To address this, I'd like to use an the AutoSizeInput component instead but it's not part of grafana ui just yet so I'll do that in a future PR. 

Screenshot from 2022-05-03 09-31-15

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/MetricsQueryEditor.test.tsx [hunk 1]
```diff
@@ -165,4 +166,50 @@ describe('QueryEditor', () => {
...
+        const result = render(
+          <MetricsQueryEditor
+            {...props}
+            query={{ ...props.query, refId: 'A', alias: 'Period: {{period}} InstanceId: {{InstanceId}}' }}
+          />
+        );
+
+        expect(screen.getByText('Label')).toBeInTheDocument();
+        expect(screen.queryByText('Alias')).toBeNull();
+        expect(result.container.querySelector('#A-cloudwatch-metric-query-editor-label')).toHaveValue(
```
Comment (sunker) [inline public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/MetricsQueryEditor.test.tsx:186]: Was not able to get this to work:
```typescript 
expect(screen.getByLabelText('Label')).toHaveValue(
          "Period: ${PROP('Period')} InstanceId: ${PROP('Dim.InstanceId')}"
        );
```

Not sure why? This is an example of the rendered HTML

```html
<div class="css-1hjz9ft">
   <label class="css-11h8yic" for="A-cloudwatch-metric-query-editor-label">
      Label<span class="css-18lesqj"> - optional</span>
      <div class="css-1vzus6i-Icon">
         <svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="14" height="14" class="css-3vp1c4">
            <path d="M12,2A10,10,0,1,0,22,12,10.01114,10.01114,0,0,0,12,2Zm0,18a8,8,0,1,1,8-8A8.00917,8.00917,0,0,1,12,20Zm0-8.5a1,1,0,0,0-1,1v3a1,1,0,0,0,2,0v-3A1,1,0,0,0,12,11.5Zm0-4a1.25,1.25,0,1,0,1.25,1.25A1.25,1.25,0,0,0,12,7.5Z"></path>
         </svg>
      </div>
   </label>
   <span class="css-djpfq6"></span>
   <div>
      <div class="css-xcstkt-input-wrapper" data-testid="input-wrapper">
         <div class="css-1w5c5dq-input-inputWrapper"><input class="css-yn255a-input-input" id="A-cloudwatch-metric-query-editor-label" placeholder="auto" value="Period: ${PROP('Period')} InstanceId: ${PROP('Dim.InstanceId')}"></div>
      </div>
   </div>
</div>
```

File: public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/Alias.tsx [hunk 2]
```diff
@@ -6,9 +6,10 @@ import { Input } from '@grafana/ui';
 export interface Props {
   onChange: (alias: any) => void;
   value: string;
+  id: string;
```
Comment (andresmgot) [inline public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/Alias.tsx:9]: nit: the `id` is usually optional

File: public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/MetricsQueryEditor.tsx [hunk 3]
```diff
@@ -130,17 +131,37 @@ export const MetricsQueryEditor = (props: Props) => {
...
-            value={preparedQuery.alias ?? ''}
-            onChange={(value: string) => onChange({ ...preparedQuery, alias: value })}
-          />
-        </EditorField>
+        {config.featureToggles.cloudWatchDynamicLabels ? (
+          <EditorField
+            label="Label"
+            width={26}
+            optional
+            tooltip="Change time series legend name using Dynamic labels. See documentation for details."
```
Comment (andresmgot) [inline public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/MetricsQueryEditor.tsx:139]: it should be possible to use a HTML element in the tooltip so you can add a link to the documentation here.

File: public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/__snapshots__/MetricsQueryEditor.test.tsx.snap [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`QueryEditor should render component 1`] = `null`;
```
Comment (andresmgot) [inline public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/__snapshots__/MetricsQueryEditor.test.tsx.snap:3]: can we avoid this snapshot test?

**Review by andresmgot — APPROVED** [review]
LGTM, just some minor comments

File: public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/MetricsQueryEditor.tsx (hunk 3 above)
Comment (sunker) [reply public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/MetricsQueryEditor.tsx:139]: Yes good point. Will do this when we have created section about this in the docs. It's the last sub task in the epic.

File: public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/__snapshots__/MetricsQueryEditor.test.tsx.snap (hunk 4 above)
Comment (sunker) [reply public/app/plugins/datasource/cloudwatch/components/MetricsQueryEditor/__snapshots__/MetricsQueryEditor.test.tsx.snap:3]: Yeah let's remove the snapshot.

**Pull request merged**